### PR TITLE
Improvements for Delta printers

### PR DESCRIPTION
**Pull request by Guilouz**
Some improvements for Delta printers is added :
- Add 2 settings in Configuration.h to define if the printer is a Delta and if it has a removable probe
- Add warning popup to inform to connect probe before to start Delta Calibration and Auto Bed Leveling
- Replaced Manual Bed Leveling (Point 1, Point 2, etc..) function which is not used with Delta by Calibration (G33) function (when Delta Printer is enabled in Configuration.h)
- Add Delta settings (Height, Segment per second, Radius, Diagonal Rod, Tower Angle Trim and Delta Endstop Adjustments) in Machine Settings
- Add one new icon for Delta Calibration

### CODE REVIEW

**Comment by oldman4U** [comment]
Please add the new button to a Comment here and mention the users who contributed interface themes so they can add the button to their themes.

Thank you

**Comment by Guilouz** [comment]
@Acenotass @AntoszHUN @bepstein111 New icon calibration.bmp is added in this PR for Delta Calibration.

**Comment by AntoszHUN** [comment]
> @Acenotass @AntoszHUN @bepstein111 New icon calibration.bmp is added in this PR for Delta Calibration.

All right. I do it in the evening after work.

File: TFT/src/User/API/Language/language_es.h [hunk 1]
```diff
@@ -49,6 +49,10 @@
     #define STRING_FWRETRACT              "FW Retraction"
     #define STRING_FWRECOVER              "FW Retraction Recover"
     #define STRING_RETRACT_AUTO           "Auto Firmware Retract"
+    #define STRING_DELTA_CONFIGURATION           "Delta Configuration"
+    #define STRING_DELTA_TOWER_ANGLE           "Tower Angle Trim"
+    #define STRING_DELTA_ENDSTOP           "Endstop Adjustments"
+    #define STRING_DIAGONAL_ROD           "Diagonal rod"
```
Comment (digant73) [inline TFT/src/User/API/Language/language_es.h:55]: This label is present only here!

File: TFT/src/User/API/MachineParameters.c [hunk 2]
```diff
@@ -37,6 +40,9 @@ const char * const parameterCode[PARAMETERS_COUNT] = {
   "M207",  // FW retract
   "M208",  // FW retract recover
   "M209",  // Set auto FW retract
+  "M665",  // Delta Configuration
```
Comment (digant73) [inline TFT/src/User/API/MachineParameters.c:43]: Param are ordered by gcode. So your gcode must be provided after M420. That logic is applied on all the files e.g. Languages, parseAck, inerfaceCmd etc._

File: TFT/src/User/Menu/BedLevelingLayer2.c [hunk 3]
```diff
@@ -73,8 +78,14 @@ void menuBedLevelingLayer2(void)
     switch (key_num)
     {
       case KEY_ICON_0:
-        if (infoMachineSettings.leveling < BL_MBL)  // if ABL
+        if (infoMachineSettings.leveling < BL_MBL && !(REMOVABLE_PROBE))  // if ABL
           ablStart();
+        else
+        if (infoMachineSettings.leveling < BL_MBL && REMOVABLE_PROBE)  // if ABL and removal probe
```
Comment (digant73) [inline TFT/src/User/Menu/BedLevelingLayer2.c:84]: put "if" on the same line of "else"

File: TFT/src/User/Menu/BedLevelingLayer2.c [hunk 4]
```diff
@@ -90,11 +101,12 @@ void menuBedLevelingLayer2(void)
         break;
 
       case KEY_ICON_3:
+        if (infoMachineSettings.zProbe == ENABLED && !(REMOVABLE_PROBE))
         infoMenu.menu[++infoMenu.cur] = menuLevelCorner;
```
Comment (digant73) [inline TFT/src/User/Menu/BedLevelingLayer2.c:105]: indent this line

File: TFT/src/User/Menu/UnifiedMove.c [hunk 5]
```diff
@@ -13,7 +18,7 @@ void menuUnifiedMove(void)
       {ICON_EXTRUDE,                 LABEL_EXTRUDE},
       {ICON_DISABLE_STEPPERS,        LABEL_DISABLE_STEPPERS},
       {ICON_BABYSTEP,                LABEL_BABYSTEP},
-      {ICON_MANUAL_LEVEL,            LABEL_LEVELING},
+      {ICON_BACKGROUND,            LABEL_BACKGROUND},
```
Comment (digant73) [inline TFT/src/User/Menu/UnifiedMove.c:21]: use here the pre compile #if.._ #else.._ the set the proper icon

File: TFT/src/User/Menu/UnifiedMove.c [hunk 6]
```diff
@@ -26,6 +31,17 @@ void menuUnifiedMove(void)
     UnifiedMoveItems.items[6].icon = ICON_LEVELING;
     UnifiedMoveItems.items[6].label.index = LABEL_BED_LEVELING;
   }
+
```
Comment (digant73) [inline TFT/src/User/Menu/UnifiedMove.c:34]: Not needed to set it at runtime. Follow my previous comment instead

**Review by digant73 — CHANGES_REQUESTED** [review]

File: TFT/src/User/API/MachineParameters.c (hunk 2 above)
Comment (Guilouz) [reply TFT/src/User/API/MachineParameters.c:43]: So M200 and M569 have wrong place no ? Problem if I put my gcode after M420, settings are not correctly applied, I have tested.

Comment (digant73) [reply TFT/src/User/API/MachineParameters.c:43]: Yes it was put in the wrong place while the others were properly ordered. I will clean up it in a further minor PR. If one of the new gcode conflicts with m420 or other gcodes it is better to identify the cause and fixing it (if possible) in parseAck only or to provide a comment highlighting the reason of different order. Also  verify that m420 continues to be properly handled if put after your gcodes. In any case the order in all the files (maybe with the exception in parseAck) can be provided

**Comment by oldman4U** [comment]
@bepstein111 Ben, when would you be available to update your theme?

**Comment by bepstein111** [comment]
> @bepstein111 Ben, when would you be available to update your theme?

Woah, my bad! Totally missed this. Will put it on the todo list for ASAP.

File: TFT/src/User/API/MachineParameters.c (hunk 2 above)
Comment (digant73) [reply TFT/src/User/API/MachineParameters.c:43]: @Guilouz Ok, but what is the entire ack message returned for M665 and M666? So we can understand the reason why M420 must be parsed after M665 and M666 (in parseAck)

**Comment by oldman4U** [comment]
bepstein111 

Any updates soon.

**Pull request merged**